### PR TITLE
fix(inputnumber): button right click bug

### PR DESCRIPTION
**Pull request by shijiatongxue**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [x] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [x] Bugfix

### PR description

Fixes InputNumber button right click bug

### Changelog
🇨🇳 Chinese
- Fix: 修复 InputNumber 按钮右键点击时数字自动增/减问题 #

---

🇺🇸 English
- Fix: fix InputNumber button right click bug

### Checklist
- [ ] Test or no need
- [ ] Document or no need
- [ ] Changelog or no need

### Other
- [ ] Skip Changelog

### Additional information

- bug video

inputNumber-right-click-bug

- [event.button MDN](https://developer.mozilla.org/zh-CN/docs/Web/API/MouseEvent/button)

image

### CODE REVIEW

File: packages/semi-foundation/inputNumber/foundation.ts [hunk 1]
```diff
@@ -317,6 +323,14 @@ class InputNumberFoundation extends BaseFoundation<InputNumberAdapter> {
...
 
+    /**
+     * Whether it is a left mouse button click
+     * @see https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button
+     */
+    _isMouseButtonLeft = (event: any) => {
+        return get(event, 'button') === numbers.MOUSE_BUTTON_LEFT;
+    }
+
     _preventDefault(event: any) {
```
Comment (pointhalo) [inline packages/semi-foundation/inputNumber/foundation.ts:334]: 这里保持一下上下书写风格吧。上下都不是箭头函数。而且foundation里基本不太需要强绑定this

Comment (shijiatongxue) [reply packages/semi-foundation/inputNumber/foundation.ts:334]: 好的

Comment (shijiatongxue) [reply packages/semi-foundation/inputNumber/foundation.ts:334]: done

**Comment by shijiatongxue** [comment]
Thanks 💖

**Pull request merged**